File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/PrepareChunkResult.java
package com.autonomi.antd.models;

import java.util.List;

/**
 * Result of preparing a single-chunk publish for external signing via
 * {@code POST /v1/chunks/prepare}.
 *
 * <p>When {@link #alreadyStored} is {@code true}, the chunk is already
 * on-network — only {@link #address} and {@link #alreadyStored} are
 * populated, and no finalize call is needed. Otherwise the wave-batch payment
 * fields describe what the external signer must submit before calling
 * {@code finalizeChunkUpload}.
 *
 * <p>Requires antd &gt;= 0.7.0.
 *
 * @param address             content-addressed BLAKE3 of the chunk bytes
 *                            (hex, 64 chars). Always set.
 * @param alreadyStored       {@code true} if the chunk is already stored and
 *                            no payment is needed.
 * @param uploadId            opaque identifier to pass back to finalize.
 *                            Empty when {@code alreadyStored == true}.
 * @param paymentType         always {@code "wave_batch"} for single-chunk
 *                            publishes. Empty when {@code alreadyStored}.
 * @param payments            per-quote payment entries for
 *                            {@code payForQuotes()}. Typically 5–7 (one per
 *                            peer in the close group). Empty when
 *                            {@code alreadyStored}.
 * @param totalAmount         total amount to pay (atto tokens, decimal
 *                            string). Empty when {@code alreadyStored}.
 * @param paymentVaultAddress payment vault contract address. Empty when
 *                            {@code alreadyStored}.
 * @param paymentTokenAddress payment token contract address. Empty when
 *                            {@code alreadyStored}.
 * @param rpcUrl              EVM RPC URL for submitting transactions. Empty
 *                            when {@code alreadyStored}.
 */
public record PrepareChunkResult(
        String address,
        boolean alreadyStored,
        String uploadId,
        String paymentType,
        List<PaymentInfo> payments,
        String totalAmount,
        String paymentVaultAddress,
        String paymentTokenAddress,
        String rpcUrl) {}

File: ant-sdk/antd-php/src/Models/PrepareChunkResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of preparing a single-chunk external-signer publish via
 * POST /v1/chunks/prepare.
 *
 * When `alreadyStored` is true, the chunk is already on-network and no
 * payment / finalize call is required — `address` is still populated but
 * the payment fields are empty strings / empty list. Otherwise the
 * wave-batch payment fields describe what the external signer must submit
 * before calling finalizeChunkUpload().
 *
 * Requires antd >= 0.7.0.
 */
readonly class PrepareChunkResult
{
    public function __construct(
        /** Content-addressed BLAKE3 of the chunk bytes (hex, 64 chars). Always set. */
        public string $address,
        /** True if the chunk is already stored on the network — no payment needed. */
        public bool $alreadyStored,
        /** Opaque identifier to pass back to finalizeChunkUpload(). Empty when alreadyStored. */
        public string $uploadId,
        /** Always "wave_batch" for single-chunk publishes. Empty when alreadyStored. */
        public string $paymentType,
        /** @var list<PaymentInfo> Per-quote payments. Empty when alreadyStored. */
        public array $payments,
        /** Total atto-token amount across all payments. Empty when alreadyStored. */
        public string $totalAmount,
        /** Payment vault contract address (hex with 0x prefix). Empty when alreadyStored. */
        public string $paymentVaultAddress,
        /** Payment token contract address (hex with 0x prefix). Empty when alreadyStored. */
        public string $paymentTokenAddress,
        /** EVM RPC URL for submitting payment transactions. Empty when alreadyStored. */
        public string $rpcUrl,
    ) {}
}
